### PR TITLE
[clkmgr, top] Hook-up clkmgr idle signals

### PR DESCRIPTION
**Pull request by tjaychen**
Auto-generate connections needed for transactional clock group idle signals.
An idle for hmac is temporarily created.  Kmac will eventually fall into this group also.

### CODE REVIEW

**Comment by tjaychen** [comment]
do you guys mind having a look? i know it's still in draft, but just wanted to get some early thoughts on hooking up the idle signals.

File: hw/ip/clkmgr/data/clkmgr.hjson.tpl [hunk 1]
```diff
@@ -127,7 +128,7 @@ num_grps = len(grps)
       swaccess: "rw",
       hwaccess: "hro",
       fields: [
-% for clk in hint_clks:
+% for clk in hint_clks.keys():
```
Comment (tjaychen) [inline hw/ip/clkmgr/data/clkmgr.hjson.tpl:131]: oops..this isn't actually needed. I'll get rid of it.

File: hw/top_earlgrey/data/top_earlgrey.hjson [hunk 2]
```diff
@@ -411,15 +411,15 @@
       'pwrmgr.pwr_rst'   : ['rstmgr.pwr'],
       'pwrmgr.pwr_clk'   : ['clkmgr.pwr'],
       'flash_ctrl.keymgr': ['keymgr.flash'],
-      'alert_handler.crashdump': ['rstmgr.alert_dump']
+      'alert_handler.crashdump': ['rstmgr.alert_dump'],
+      'clkmgr.idle'      : [],
```
Comment (tjaychen) [inline hw/top_earlgrey/data/top_earlgrey.hjson:415]: leave a comment here to indicate that `clkmgr.idle` is auto populated by the merge function.

**Review by msfschaffner — COMMENTED** [review]
This looks good from my side. Would OTP use the same connection to indicate idle write status?

**Comment by tjaychen** [comment]
ah otp doesn't need this for the time being.. since this is most an idle
indication needed for the transactional clock group.

File: util/topgen.py [hunk 3]
```diff
@@ -532,12 +532,20 @@ def generate_clkmgr(top, cfg_path, out_path):
...
-    hint_clks = OrderedDict([
+    hints = OrderedDict([
         (clk, src)
         for grp in grps for (clk, src) in grp['clocks'].items()
         if grp['sw_cg'] == 'hint' and not src_aon_attr[src]
     ])
 
+    # hint clocks dict
+    for clk, src in hints.items():
+        # the clock if constructed as clk_{src_name}_{module_name}.
```
[the remark this answers is not in the document]
Comment (tjaychen) [reply util/topgen.py]: fixed.

**Review by msfschaffner — APPROVED** [review]

**Pull request merged**